addons/survey/models/survey_survey.py

```diff
@@ -678,6 +678,44 @@ def _get_conditional_maps(self):
                     triggered_questions_by_answer[question.triggering_answer_id] = question
         return triggering_answer_by_question, triggered_questions_by_answer
 
+    # ------------------------------------------------------------
+    # BACKGROUND MANAGEMENT
+    # ------------------------------------------------------------
+    def _get_section_info_by_question(self):
+        """
+        Prepares a dict that will be used to handle background transitions
+        For each survey question, we return
+            survey_question_id : {
+                - section_id : section of the current question
+                - next_question_id : the question that follow the current one (no conditional check here)
```

Comment:
A bit meta, but isn't it yet another "next question computation inlined in code" ? Becoming lost in all those next questions stuff ...

Reply:
the next question method is computing the next question to display based on the context (user answers, section, conditional configuration, etc..), here we just want to sent to the frontend the survey structure, no matter the conditional configuration. This is needed to compute the next question to display at frontend side (to find the next section to display, depending on conditional configuration).
So at the end, we have a "get_next_question" method at backend side and one at frontend side. Both should be aligned. + we have one in session manager as the logic is a bit different here (as we base our next question on most_voted_answers)